universal-login-relayer/lib/utils/utils.js

```diff
@@ -42,6 +44,18 @@ const withENS = (provider, ensAddress) => {
   return new providers.Web3Provider(provider._web3Provider, chainOptions);
 };
 
+const isEnoughGasLimit = (estimateGas, gasLimit) =>
+  gasLimit >= estimateGas;
+
+const hasEnoughToken = async (gasToken, identityAddress, gasLimit, provider) => {
+  if (gasToken !== ether) {
+    const token = new ethers.Contract(gasToken, ERC20.interface, provider);
+    const identityTokenBalance = await token.balanceOf(identityAddress);
+    return identityTokenBalance >= gasLimit;
+  } 
+  return true;
```

Comment:
If not implemented throw informative exception i.e. `NotYetImplemented('Ether refunds are not yet supported').

Reply:
✅